lib/src/parser/recursive.dart

```diff
@@ -55,67 +53,102 @@ class RecursiveAstParser {
   }
 
   /// Parses and returns a template AST beginning at the token provided.
+  /// No desugaring of any kind occurs here.
   TemplateAst parseDecorator(NgToken beginToken) {
     // The first token is the decorator/name.
-    final nameToken = _reader.expect(NgTokenType.elementDecorator);
-    final startChar = nameToken.lexeme.codeUnitAt(0);
-    NgToken valueToken;
-    NgToken endValueToken;
-    var endToken = nameToken;
+    NgToken prefixToken;
+    NgToken decoratorToken;
+    NgToken suffixToken;
 
-    // If the next token starts a value then we need to parse the expression.
-    if (_reader.peek().type == NgTokenType.beforeElementDecoratorValue) {
-      _reader.next();
-      valueToken = _reader.expect(NgTokenType.elementDecoratorValue);
-      endValueToken = _reader.expect(NgTokenType.afterElementDecoratorValue);
-      endToken = endValueToken;
+    NgTokenType peekType = _reader.peekType();
+    if (peekType == NgTokenType.bananaPrefix ||
+        peekType == NgTokenType.eventPrefix ||
+        peekType == NgTokenType.propertyPrefix) {
+      prefixToken = _reader.next();
+      decoratorToken = _reader.next();
+      suffixToken = _reader.next();
+    } else if (peekType == NgTokenType.referencePrefix ||
+        peekType == NgTokenType.templatePrefix) {
+      prefixToken = _reader.next();
+      decoratorToken = _reader.next();
+    } else {
+      decoratorToken = _reader.next();
     }
 
-    // Determine what the decorator type is, i.e. either a:
-    // - property
-    // - event
-    // - reference
-    // - attribute
-    switch (startChar) {
-      case $open_bracket:
-        return new PropertyAst.parsed(
+    NgAttributeValueToken valueToken;
+    NgToken equalSignToken;
+
+    if (_reader.peekTypeIgnoringType(NgTokenType.whitespace) ==
```

Comment:
Yes at the current moment. However, I think we can change this to not allow whitespaces in between - which would make the parsing a lot easier at this step too. @matanlurey  what do you think?